MeshLib/Node.h

```diff
@@ -70,6 +70,9 @@ class Node : public MathLib::Point3dWithID
         return Node(_x[0]-v[0], _x[1]-v[1], _x[2]-v[2]);
     }
 
+    /// Sets the ID of a node to the given value.
+    void setID(std::size_t id) { this->_id = id; }
```

Comment:
You can use one of the `Node` constructors to set the `id`.  Why do you need this method?